src/tls/cockpit-certificate-helper.in

```diff
@@ -35,11 +36,15 @@ ipa service-show "${SERVICE}" || ipa service-add --ok-as-delegate=true --force "
 mkdir -p /etc/cockpit
 klist -k /etc/cockpit/krb5.keytab | grep -qF "${SERVICE}" || ipa-getkeytab -p "HTTP/${HOST}" -k /etc/cockpit/krb5.keytab
 
-# request an SSL certificate; be sure to not leave traces of the .key on disk or
-# get race conditions with file permissions; also, ipa-getcert
-# cannot directly write into /etc/cockpit due to SELinux
+# ipa-getcert cannot set permisisons nor write into /etc/cockpit due to SELinux
 if ipa-getcert request -f /run/cockpit/ipa.crt -k /run/cockpit/ipa.key -K "HTTP/${HOST}" -w -v; then
-    mv /run/cockpit/ipa.crt /etc/cockpit/ws-certs.d/10-ipa.cert
-    cat /run/cockpit/ipa.key  >> /etc/cockpit/ws-certs.d/10-ipa.cert
-    rm -f /run/cockpit/ipa.key
+    mv -Z /run/cockpit/ipa.crt /etc/cockpit/ws-certs.d/10-ipa.cert
+    mv -Z /run/cockpit/ipa.key /etc/cockpit/ws-certs.d/10-ipa.key
+    rm -f /run/cockpit/ipa.key /run/cockpit/ipa.crt
```

Comment:
Given that the script is `set -e`, can this ever happen? If `mv` succeeds, the original file should be gone already, no?

Reply:
ah.  true.  i copied this over from the merging code without thinking about it too deeply :)